File: server/src/main/java/umm3601/wordRiver/Sentence.java
package umm3601.wordRiver;
import org.mongojack.Id;
import org.mongojack.ObjectId;
import java.util.List;

public class Sentence {
    @ObjectId @Id
    // sentence mongo object Id
    public String mongoObjectId;
    public String sentenceId;
    // the sentence contents
    public String sentenceText;
    // the time the sentence was submitted in storybuilder
    public String timeSubmitted;
    // the object id for the learner that submitted the sentence
    public String learnerId;
    public List<Word> words;
    public List<String> selectedWordForms;
    // mongo object Id of associated user
    public String userId;
    public List<String> contextPackIds;
}

File: client/src/app/datatypes/sentence.ts
import { Word } from './word';

export class Sentence {
  sentenceId: string;
  sentenceText: string;
  timeSubmitted: string;
  learnerId: string;
  words: Array<Word>;
  selectedWordForms: Array<string>;
  userId: string;
  contextPackIds: Array<string>;
  wordCountPairs: { word: string; count: number }[];
  repeatedWords: { word: string; count: number }[];
  uniqueWords: { word: string; count: number }[] = []; // New property for unique words
  uniqueWordCount: number; // For total count of unique words
  wordCountMap: Record<string, number>;

  constructor() {
    this.repeatedWords = [];
    this.wordCountPairs = [];
    this.uniqueWords = []; // Initialize uniqueWords array
    this.uniqueWordCount = 0;
  }
}
